contracts/USM.sol

```diff
@@ -1,117 +1,138 @@
+        require(_usmToBurn >= MIN_BURN_AMOUNT, "Must burn at least 1 USM");
+        uint ethToSend = _burn(_usmToBurn);
+        require(debtRatio() <= WAD,
+            "Cannot burn with debt ratio below 100%");
+        Address.sendValue(msg.sender, ethToSend);
+        // set latest fum price
+        _setLatestFumPrice();
+        return (ethToSend);
     }
 
     /**
-     * @notice Calculate debt ratio of the current Eth pool amount and outstanding USM
-     * (the amount of USM in total supply).
-     *
-     * @return Debt ratio in UNIT.
+     * @notice Funds the pool with ETH, minting FUM at its current price and considering if the debt ratio goes from under to over
      */
-    function debtRatio() external view returns (uint) {
-        if (ethPool == 0) {
-            return 0;
+    function fund() external payable {
+        require(msg.value > MIN_ETH_AMOUNT, "Must deposit more than 0.001 ETH");
+        if(debtRatio() > MAX_DEBT_RATIO){
+            uint ethNeeded = _usmToEth(totalSupply()).wadDiv(MAX_DEBT_RATIO).sub(ethPool).add(1); //+ 1 to tip it over the edge
+            if (msg.value >= ethNeeded) { // Split into two fundings at different prices
+                _fund(msg.sender, ethNeeded);
+                _fund(msg.sender, msg.value.sub(ethNeeded));
+                return;
+            } // Otherwise continue for funding the total at a single price
         }
-        // If divFixed is fed two integers, returns their division as a fixed point number
-        return divFixed(totalSupply(), _ethToUsm(ethPool));
+        _fund(msg.sender, msg.value);
     }
 
     /**
-     * @notice Convert ETH amount to USM using the latest price of USM
-     * in ETH.
-     *
-     * @param _ethAmount The amount of ETH to convert.
-     * @return The amount of USM.
+     * @notice Funds the pool with ETH, minting FUM at its current price
      */
-    function _ethToUsm(uint _ethAmount) internal view returns (uint) {
-        require(_ethAmount > 0, "Eth Amount must be more than 0");
-        return mulFixed(_oraclePrice(), _ethAmount);
+    function _fund(address to, uint ethIn) internal {
+        uint _fumPrice = fumPrice();
+        uint fumOut = ethIn.wadDiv(_fumPrice);
+        ethPool = ethPool.add(ethIn);
+        fum.mint(to, fumOut);
+        _setLatestFumPrice();
     }
 
     /**
-     * @notice Convert USM amount to ETH using the latest price of USM
-     * in ETH.
-     *
-     * @param _usmAmount The amount of USM to convert.
-     * @return The amount of ETH.
+     * @notice Defunds the pool by sending FUM out in exchange for equivalent ETH
+     * from the pool
      */
-    function _usmToEth(uint _usmAmount) internal view returns (uint) {
-        require(_usmAmount > 0, "USM Amount must be more than 0");
-        return divFixed(_usmAmount, _oraclePrice());
+    function defund(uint _fumAmount) external {
+        require(_fumAmount >= WAD, "Must defund at least 1 FUM");
```

Comment:
I think we should just remove this line.  1 FUM could in principle ending up being pretty valuable!  And I don't see any reason not to let them redeem 0.000001 FUM if that's how they want to waste their gas.